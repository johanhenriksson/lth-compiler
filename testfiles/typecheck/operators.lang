int main() {
    int a = 5 + 3;
    int b = 5 + true; // 2x error
    int c = true + 5; // 2x error

    a = 5 - 3 * 3 + 1 % 3;
}
