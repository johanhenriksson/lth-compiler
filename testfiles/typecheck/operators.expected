Error at line 3,5: cannot assign 'b' of type int, got <unknown>
Error at line 3,13: mismatched operands int and bool
Error at line 4,5: cannot assign 'c' of type int, got <unknown>
Error at line 4,13: mismatched operands bool and int
